### ISSUE TITLE
`make install` fails for 8.13

### ISSUE THREAD

**Issue by shamefulCake1** [racket/racket#4999]
**What version of Racket are you using?**
8.13

**What program did you run?**
```
../configure   --prefix=/usr   --libdir=/usr/lib64   --mandir=/usr/man   --datadir=/usr/share/racket   --docdir=/usr/doc/racket-8.13   --build=x86_64-slackware-linux
make -j4
make install DESTDIR=/tmp/package-racket
```
**What should have happened?**
Install should have passed without an error.

**If you got an error message, please include it here.**
```
Fixing package links at: /tmp/racket/usr/share/racket/racket/links.rktd...
*** Error: undoing changes...
regexp-match: contract violation
  expected: (or/c bytes? string? input-port? path?)
  given: '(#"pkgs" #"base")
  context...:
   /usr/sbo/repo/development/racket/racket-8.13/collects/setup/unixstyle-install.rkt:322:5
   /usr/sbo/repo/development/racket/racket-8.13/collects/setup/unixstyle-install.rkt:583:2
   body of (submod "/usr/sbo/repo/development/racket/racket-8.13/collects/setup/unixstyle-install.rkt" main)
   body of "/usr/sbo/repo/development/racket/racket-8.13/collects/setup/main.rkt"
failed
 in build-one
 in loop
 in module->hash
make: *** [Makefile:25: install] Error 1
```

**Please include any other relevant details**
x86_64-slackware-linux 15.0

**Comment by mflatt** [comment]
Thank you for the report!

Short answer: If you can use `--datadir=/usr/share`, that's probably what you want, because `racket` is added anywhere that the `datadir` configuration is used. That will turn out to avoid the problem.

Another possibility is to add `--pkgsdir=/usr/share/racket/racket/pkgs`, which keeps the package directory at the expected location relative to `datadir`, and so also avoids a bug in adjustments for a moved package directory.


Overall, it looks like there are two problems:

* The default location for packages is derived from `datarootdir` instead of `datadir`, which seems wrong, but it must have been that way always.
* The installation code to deal with a package directory not at the usual place (relative to `datadir`) has been broken since v8.2 (July 2021).

I plan to fix both of those.

**Comment by shamefulCake1** [comment]
In fact, I worked around this by just not specifying `--datadir` at all, and shared data still went to the directory I expected.

But the problem there is really not just about the directory, but about the `fix-pkg-links` procedure, which is receiving wrong arguments.

**Comment by mflatt** [comment]
> But the problem there is really not just about the directory, but about the fix-pkg-links procedure, which is receiving wrong arguments.

Right, thanks — that's what the second bullet above is referring to (i.e., the bug since v8.2).
